File: src/main/java/ru/ezhov/rocket/action/properties/ResourceGeneralPropertiesRepository.java
package ru.ezhov.rocket.action.properties;

import ru.ezhov.rocket.action.notification.NotificationFactory;
import ru.ezhov.rocket.action.notification.NotificationType;

import java.io.IOException;
import java.util.Properties;

public class ResourceGeneralPropertiesRepository implements GeneralPropertiesRepository {
    @Override
    public Properties all() {
        Properties properties = new Properties();
        try {
            properties.load(this.getClass().getResourceAsStream("/general.properties"));
        } catch (IOException e) {
            e.printStackTrace();

            NotificationFactory.getInstance().show(NotificationType.ERROR, "Error read general properties");
        }
        return properties;
    }
}

File: src/main/kotlin/ru/ezhov/rocket/action/properties/ResourceGeneralPropertiesRepository.kt
package ru.ezhov.rocket.action.properties

import ru.ezhov.rocket.action.notification.NotificationFactory
import ru.ezhov.rocket.action.notification.NotificationType
import java.io.IOException
import java.util.*

class ResourceGeneralPropertiesRepository : GeneralPropertiesRepository {
    override fun all(): Properties {
        val properties = Properties()
        try {
            properties.load(this.javaClass.getResourceAsStream("/general.properties"))
        } catch (e: IOException) {
            e.printStackTrace()
            NotificationFactory.notification.show(NotificationType.ERROR, "Error read general properties")
        }
        return properties
    }
}